### PR TITLE
Fix for User-Agent header duplication on each http request

### PR DESCRIPTION
**Pull request by daletskyi**
Currently "User-Agent" header is added on each `MakeRequest` call and it's value gets duplicated ([docs](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.headers.httpheaders.add?view=netcore-2.0#System_Net_Http_Headers_HttpHeaders_Add_System_String_System_String_)). If client application uses `Analytics.Client` as a singleton then after some time "User-Agent" header reaches request limits on Segment and it will always return `400 Bad Request`. This PR moves header initilialization to `BlockingRequestHandler`constructor and also fixes handling of errors.

### CODE REVIEW

**Comment by f2prateek** [comment]
Thanks! This LGTM at a high level - @kevingilliard can you help review this?

File: Analytics/Request/BlockingRequestHandler.cs [hunk 1]
```diff
@@ -243,7 +249,7 @@ internal BlockingRequestHandler(Client client, TimeSpan timeout)
 							// If status code is greater than 500 and less than 600, it indicates server error
 							// Error code 429 indicates rate limited.
 							// Retry uploading in these cases.
-							Task.Delay(backoff).Wait();
+							await Task.Delay(backoff).ConfigureAwait(false);
```
Comment (f2prateek) [inline Analytics/Request/BlockingRequestHandler.cs]: Why did this change?

File: Analytics/Request/BlockingRequestHandler.cs [hunk 2]
```diff
@@ -257,7 +263,7 @@ internal BlockingRequestHandler(Client client, TimeSpan timeout)
 #endif
 				}
 
-				if (backoff == MAXIMUM_BACKOFF_DURATION && statusCode != (int)HttpStatusCode.OK)
+				if (backoff >= MAXIMUM_BACKOFF_DURATION || statusCode != (int)HttpStatusCode.OK)
```
Comment (f2prateek) [inline Analytics/Request/BlockingRequestHandler.cs]: Can you explain this change?

File: Analytics/Request/BlockingRequestHandler.cs [hunk 3]
```diff
@@ -165,12 +164,19 @@ internal BlockingRequestHandler(Client client, TimeSpan timeout)
...
 				{
 					{ "batch id", batch.MessageId },
 					{ "json size", json.Length },
 					{ "batch size", batch.batch.Count }
-				});
+				};
+
+				if (_client.Config.CompressRequest)
+				{
+					requestInfo.Add("gzipped json size", requestData.Length);
```
Comment (f2prateek) [inline Analytics/Request/BlockingRequestHandler.cs]: This seems unrelated?

File: Analytics/Request/BlockingRequestHandler.cs (hunk 1 above)
Comment (daletskyi) [reply Analytics/Request/BlockingRequestHandler.cs]: Since the `MakeRequest` method is async using `Wait()` instead of `await` can cause deadlock. I can move it to another PR if needed, just thought it is too small for separate one.

File: Analytics/Request/BlockingRequestHandler.cs (hunk 3 above)
Comment (daletskyi) [reply Analytics/Request/BlockingRequestHandler.cs]: Right, will remove it.

File: Analytics/Request/BlockingRequestHandler.cs (hunk 2 above)
Comment (daletskyi) [reply Analytics/Request/BlockingRequestHandler.cs]: `backoff` is changed only after delay where it is multiplied by 2 each time. The initial value is 100, so it will be 200, 400, 800 and so on. `MAXIMUM_BACKOFF_DURATION` is set to 10000 so if statement will never be executed. The second part will work in the next situation: if you get response 400 before `backoff` reached max value then you need to check status code here.

File: Analytics/Request/BlockingRequestHandler.cs (hunk 1 above)
Comment (f2prateek) [reply Analytics/Request/BlockingRequestHandler.cs]: I'd like to be to seperate if you don't mind! It makes it easy to independently revert the header change from this one if needed and also link/attribute properly from the changelog.

File: Analytics/Request/BlockingRequestHandler.cs (hunk 2 above)
Comment (f2prateek) [reply Analytics/Request/BlockingRequestHandler.cs]: Got it - could this be a different PR similar to https://github.com/segmentio/Analytics.NET/pull/83/files#r197556500

File: Test.Net45/Test.Net45.csproj [hunk 4]
```diff
@@ -36,8 +36,8 @@
     <Prefer32Bit>false</Prefer32Bit>
   </PropertyGroup>
   <ItemGroup>
-    <Reference Include="nunit.framework, Version=3.9.0.0, Culture=neutral, PublicKeyToken=2638cd05610744eb, processorArchitecture=MSIL">
-      <HintPath>..\packages\NUnit.3.9.0\lib\net45\nunit.framework.dll</HintPath>
+    <Reference Include="nunit.framework, Version=3.8.1.0, Culture=neutral, PublicKeyToken=2638cd05610744eb, processorArchitecture=MSIL">
```
Comment (f2prateek) [inline Test.Net45/Test.Net45.csproj]: Curious - why did this change?

**Comment by kevingilliard** [comment]
Hi, @f2prateek 
What can I do for this ticket?

**Comment by f2prateek** [comment]
@kevingilliard take a look at the changes and let me or @daletskyi know if you spot any issues with this PR or if it looks good to you :)

File: Test.Net45/Test.Net45.csproj (hunk 4 above)
Comment (daletskyi) [reply Test.Net45/Test.Net45.csproj]: I couldn't build test project because version 3.8.1 was referenced in [packages.config](https://github.com/segmentio/Analytics.NET/blob/master/Test.Net45/packages.config#L3). Since the same version is used across all other test projects I just fixed the path here.
